sdk/js/src/DefaultCacheStore.ts

```diff
@@ -1,26 +1,41 @@
-export class DefaultCacheStore {
-    private store?: Storage
+// Abstract Storage class
+import { DVCStorage } from './types'
+import { checkIsServiceWorker } from './utils'
 
-    constructor() {
-        this.store =
-            typeof window !== 'undefined'
-                ? window.localStorage
-                : stubbedLocalStorage
+export abstract class StorageStrategy implements DVCStorage {
+    abstract store: unknown
+    abstract init(): Promise<unknown>
+    abstract save(storeKey: string, data: unknown): Promise<void>
+    abstract load<T>(storeKey: string): Promise<T | undefined>
+    abstract remove(storeKey: string): Promise<void>
+}
+
+// LocalStorage implementation
+export class LocalStorageStrategy extends StorageStrategy {
+    override store: Storage
+    private isTesting: boolean
+
+    constructor(isTesting = false) {
+        super()
+        this.isTesting = isTesting
+        this.init()
     }
 
-    save(storeKey: string, data: unknown): void {
-        this.store?.setItem(storeKey, JSON.stringify(data))
+    async init(): Promise<void> {
+        this.store = this.isTesting ? stubbedLocalStorage : window.localStorage
```

Comment:
does this need to be in runtime code? seems like it should be mocked by the tests?

Reply:
not sure, this is how it was done before

Reply:
will see if its easy enough to change

Reply:
is it alright if I make this a separate ticket? Since it was already being done this way I'd rather not mess with it in an already risky pr

Reply:
Made it into a ticket in our triage, Ill try and pick it up during my oncall shift next week